core/trino-main/src/main/java/io/trino/server/testing/TestingSystemSecurityMetadata.java

```diff
@@ -221,7 +238,29 @@ public void setTableOwner(Session session, CatalogSchemaTableName table, TrinoPr
     @Override
     public Optional<Identity> getViewRunAsIdentity(Session session, CatalogSchemaTableName viewName)
     {
-        return Optional.empty();
```

Comment:
Why do you need to change this?

Reply:
This method is used to determine the actual view owner's identity, in context of the currently enabled security system. Returning `empty()` from here will make the analyzer create a bare-bones identity with no enabled roles.

Reply:
It is testing metadata, can we simply inject an owner?

Reply:
It's kind of simpler to just implement this method :)

Reply:
The current implementation it is not necessarly the correct way of getting the owner. It may differ between access controls. Injecting is always correct for tests as leave it for a test and it is possible to test different scenarios.